### DISCUSSION TITLE
[A/L] (2021/22) Induzione - Esercizio 19

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#319] [Esercizi A-L]
![image](https://user-images.githubusercontent.com/69389461/152323279-cb6a4fa8-8c19-46df-9673-7a61818d3726.png)

**Comment by Exyss** [comment]
L'errore sta nel passo algebrico all'interno del passo induttivo:

- L'identità proposta è $3 \cdot (n+1) = 3 \cdot (a + -b)$. Possiamo riscrivere tale identità come $3 \cdot n + 3 \cdot 1 = 3 \cdot a + 3 \cdot b$.

- Assumendo che l'ipotesi sia vera, a questo punto otterremmo che $3 \cdot  n = 3 \cdot a = 3 \cdot b = 0$, dunque l'identità viene riscritta come $0 + 3 \cdot 1 = 0 + 0$, ottenendo che $3 = 0$, cosa che è matematicamente impossibile.

**Comment by Elia-Belli** [comment]
Nel passo induttivo applicando l'ipotesi induttiva si arriva ad una contraddizione.

Abbiamo definito $n+1 = a+b$ con $0 \leq a,b \in \mathbb N \< n$.
Siccome a e b sono minori di n possiamo applicare $P(n)$: $3 \times a = 0$ e $3 \times b = 0$, ma sappiamo che il prodotto di due numeri naturali è 0 solo quando almeno uno è 0, in questo caso $3 \not= 0 \Rightarrow a,b=0$.
Ma allora $n+1 = a+b = 0+0$ che è falso $\forall n \geq 0$

**Comment by LuginiAndrea** [comment]
Per quanto entrambe le soluzioni già proposte siano giuste, direi che il vero problema è di natura induttiva.
Infatti nel passo induttivo assumiamo che $n + 1 = a + b, 0 \leq a,b \leq n$. 
Il problema sta nell'assumere che $1 \in \left[0,n\right]$, il che non è vero nel caso $n = 0$
